### PR TITLE
Move usePreventLeavingPage to navigation package

### LINKED ISSUE #36723: Move usePreventLeavingPage to navigation package
The `usePreventLeavingPage` hook currently exists under `client/hooks/`.  This is a useful hook that might be best moved under `@woocommerce/navigation` for reuse.

### PR DESCRIPTION
**Pull request by joshuatf**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

* Moves the `usePreventLeavingPage` hook to the navigation package
* Renames the hook to `useConfirmUnsavedChanges`

Closes #36723 .

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Navigate to the new product experience
2. Make some changes to the form
3. Click on a different menu item in the admin menu
4. Check that you are prompted to save before leaving

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by louwie17 — DISMISSED** [review]
Thanks for moving this over @joshuatf this tested well and changes look good!
This will require a re-approval after the conflicts have been solved.

**Comment by louwie17** [comment]
@joshuatf looks like it needs a rebase again and the tests action is failing 😢 
Feel free to ping me in Slack once updated so we can get this merged quicker :)

File: packages/js/navigation/src/index.js
Comment (joshuatf) [inline packages/js/navigation/src/index.js:21]: This was causing an issue with mocking.  Not sure why it was needed for tests, but all tests are passing without it.

**Comment by joshuatf** [comment]
@louwie17 This should be good for a review now.

File: packages/js/navigation/src/index.js
Comment (louwie17) [reply packages/js/navigation/src/index.js:21]: Yeah that seems rather strange on why this was suppose to happen.

**Review by louwie17 — APPROVED** [review]
LGTM 🚀

**Comment by louwie17** [comment]
Merged it for ya :)

**Pull request merged**
